src/hooks/useUnipoolRewards.ts

```diff
@@ -40,5 +43,35 @@ export default function useUnipoolRewards() {
     }
   }, [account, mounted, unipoolContract])
 
-  return [earned, rewardsLink]
+  useEffect( () => {
+    if (!unipoolContract || !account) {
+      return
+    }
+
+    const fetchRewardAPY = async () => {
+      try {
+        const rewardRateResult = await unipoolContract.rewardRate()
+        // Contract value is bn.js so we need to convert it to bignumber.js
+        const rewardRate = new BigNumber(rewardRateResult.toString())
+
+        const totalSupplyResult = await unipoolContract.totalSupply()
+        // Contract value is bn.js so we need to convert it to bignumber.js
+        const totalSupply = new BigNumber(totalSupplyResult.toString())
+
+        const rewardAPYRaw = rewardRate.multipliedBy(DURATION * 12).dividedBy(totalSupply)        
+        const rewardAPYFormatted = rewardAPYRaw.multipliedBy(100).toFixed(2)+"%"
```

Comment:
This hook just needs to return the raw value without being formatted, the responsibility of how to display the data and format it is from the ui component so here i would just do delete the `rewardAPYFormatted`